src/libcore/mem.rs

```diff
@@ -548,8 +548,19 @@ pub unsafe fn uninitialized<T>() -> T {
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub fn swap<T>(x: &mut T, y: &mut T) {
-    unsafe {
-        ptr::swap_nonoverlapping(x, y, 1);
+    if size_of::<T>() < 32 {
```

Comment:
32 is the [block size](https://github.com/rust-lang/rust/blob/master/src/libcore/ptr.rs#L167) used in `ptr::swap_nonoverlapping_bytes`, so is the point where the simd loop there starts to potentially do something interesting.

That choice is from PR https://github.com/rust-lang/rust/pull/40454, which has history and a bunch of perf numbers.